### PR TITLE
PFP Validation Modules

### PR DESCRIPTION
**Pull request by henrylay97**
Adding Ed's PFP Validation Modules and the all the plotting machinery to go with them. Has been tested and used in [validating](https://dbweb9.fnal.gov:8443/TestCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/6202&platform=Linux%203.10.0-1160.11.1.el7.x86_64&phase=ci_validation&buildtype=slf7%20e19:prof) a reconstruction change. Part of an effort to combine the usage of the different reconstruction validation modules into one consolidated workflow.

### CODE REVIEW

File: sbnci/scripts/pfpslicevalidation.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+## Allows for different naming structure in cfg file
+export ref_sbndcode_ana_hist=$(echo ${ref_pfpslice_hist})
```
Comment (vitodb) [inline sbnci/scripts/pfpslicevalidation.sh:4]: I think here 
`export ref_sbndcode_ana_hist=${ref_pfpslice_hist}`
should just work

File: sbnci/scripts/pfpvalidation.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+## Allows for different naming structure in cfg file
+export ref_sbndcode_ana_hist=$(echo ${ref_pfp_hist})
```
Comment (vitodb) [inline sbnci/scripts/pfpvalidation.sh:4]: Also here I think
`export ref_sbndcode_ana_hist=${ref_pfp_hist}`
should just work

**Review by vitodb — COMMENTED** [review]
I also noted a typo I introduced in the first line of thresholds files, it should read
`# this threshold file uses # as field separator`
i.e. filed -> field

I'll let Ed comment on changes to `sbnci/Modules/`

File: sbnci/scripts/pfpslicevalidation.sh (hunk 1 above)
Comment (henrylay97) [reply sbnci/scripts/pfpslicevalidation.sh]: Very true, I have updated!

**Comment by henrylay97** [comment]
Thanks for taking a look so quickly Vito! I have updated the bash scripts and will update the threshold files for **all** validation wfs in a separate PR.

**Comment by etyley** [comment]
Is this PR based on my branch of a similar name? I can see you have added the same files but can't see if there are any changes with respect to that branch. If it is, could you change to base of what this PR is being compared against?

**Comment by henrylay97** [comment]
@etyley yep sorry it is. The only commit that affects your stuff is [this](https://github.com/SBNSoftware/sbnci/pull/6/commits/9984fc78bd0f99ab262521e7564b8fe7958f913d) the rest was just so I had all of the plotting machinery in one place :)

**Comment by etyley** [comment]
Okay that looks good to me. The only comment would be to maybe put it inside a fcl controllable if statement in case anyone wants to ever disable it for whatever reason.

**Comment by henrylay97** [comment]
That's a good shout I will do so. Thanks v much for taking a look! :)

**Review by etyley — APPROVED** [review]

**Pull request merged**